pkg/specgen/generate/container_create.go

```diff
@@ -535,32 +535,68 @@ func createContainerOptions(ctx context.Context, rt *libpod.Runtime, s *specgen.
 	return options, nil
 }
 
-func Inherit(infra libpod.Container) (opts []libpod.CtrCreateOption, infraS *spec.Spec, compat *libpod.InfraInherit, err error) {
+func Inherit(infra libpod.Container, s *specgen.SpecGenerator, rt *libpod.Runtime) (opts []libpod.CtrCreateOption, infraS *spec.Spec, compat *libpod.InfraInherit, err error) {
+	inheritSpec := &specgen.SpecGenerator{}
+	_, err = ConfigToSpec(rt, inheritSpec, infra.ID())
+	if err != nil {
+		return nil, nil, nil, err
+	}
 	options := []libpod.CtrCreateOption{}
 	compatibleOptions := &libpod.InfraInherit{}
 	infraConf := infra.Config()
 	infraSpec := infraConf.Spec
 
-	config, err := json.Marshal(infraConf)
+	// multiple unmarshals/marshals, first marshal each part of the infraSpecGen we need
+	// then unmarshal into the compat options and THEN into our current specgen. We do not want all of the options only some
+	specGSec, err := json.Marshal(inheritSpec.ContainerSecurityConfig)
 	if err != nil {
 		return nil, nil, nil, err
 	}
-	err = json.Unmarshal(config, compatibleOptions)
+
+	specGStore, err := json.Marshal(inheritSpec.ContainerStorageConfig)
 	if err != nil {
 		return nil, nil, nil, err
 	}
-	if infraSpec.Linux != nil && infraSpec.Linux.Resources != nil {
-		resources, err := json.Marshal(infraSpec.Linux.Resources)
-		if err != nil {
-			return nil, nil, nil, err
-		}
-		err = json.Unmarshal(resources, &compatibleOptions.InfraResources)
-		if err != nil {
-			return nil, nil, nil, err
-		}
+
+	specgRes, err := json.Marshal(inheritSpec.ContainerResourceConfig)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	err = json.Unmarshal(specGSec, compatibleOptions)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	err = json.Unmarshal(specGStore, compatibleOptions)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	err = json.Unmarshal(specgRes, compatibleOptions)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	var compatByte []byte
+	compatByte, err = json.Marshal(compatibleOptions)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	err = json.Unmarshal(compatByte, &s.ContainerSecurityConfig)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	err = json.Unmarshal(compatByte, &s.ContainerStorageConfig)
+	if err != nil {
+		return nil, nil, nil, err
```

Comment:
This looks absolutely insane, why do we need that many marshal/unmarshal calls? It is completely unreadable and nobody will understand what this is doing.

Why can't we just assign the options we need instead of using marshal/unmarshal?

Reply:
i will try to find a more succinct form of this. It needs to be unmarshall unless we want a function that is basically just a list of 100 assignments

Reply:
fixed it up a bit @Luap99, lets see if it works...